File: backend/app/schemas/agent.py
from typing import List, Dict, Any, Optional
from pydantic import BaseModel

class ToolExecutionDTO(BaseModel):
    tool: str
    args: Dict[str, Any]
    result: Any

class AgentQueryRequestDTO(BaseModel):
    question: str
    max_steps: int = 5

class AgentQueryResponseDTO(BaseModel):
    answer: str
    trace: List[ToolExecutionDTO]
    using_mock: bool

File: apps/merchant-portal/src/types/agent.ts
export interface ToolExecutionDTO {
  tool: string;
  args: Record<string, any>;
  result: any;
}

export interface AgentQueryRequestDTO {
  question: string;
  max_steps?: number;
}

export interface AgentQueryResponseDTO {
  answer: string;
  trace: ToolExecutionDTO[];
  using_mock: boolean;
}
